e2e/capacity/provider_boost.test.ts

```diff
@@ -53,4 +62,140 @@ describe('Capacity: provider_boost extrinsic', function () {
     await assert.doesNotReject(boostProvider(fundingSource, booster, provider, 1n * DOLLARS));
     await assert.doesNotReject(boostProvider(fundingSource, booster, provider, 1n * DOLLARS));
   });
+
+  it('can boost, claim, and unstake via proxy account', async function () {
+    const boostAmount = 2n * DOLLARS;
+
+    const alice = await createAndFundKeypair(fundingSource, 1000n * DOLLARS, 'alice');
+    const bob = await createAndFundKeypair(fundingSource, 2000n * DOLLARS, 'bob');
+
+    const stakeKeys = createKeys('booster');
+    const provider = await createMsaAndProvider(fundingSource, stakeKeys, 'Provider1', providerBalance);
+
+    await addProxy(bob, getUnifiedAddress(alice), 'Staking');
+    const innerBoost = ExtrinsicHelper.api.tx.capacity.providerBoost(provider, boostAmount);
+    const expectedEvent = ExtrinsicHelper.api.events.capacity.ProviderBoosted;
+    await ExtrinsicHelper.proxySignAndSend(innerBoost, alice, bob, expectedEvent);
+
+    // Verify that the stakingLedger reflects that bob is staking `boostAmount`
+    const stakingLedgerRes = await ExtrinsicHelper.apiPromise.query.capacity.stakingAccountLedger(
+      getUnifiedAddress(bob)
+    );
+    assert(stakingLedgerRes.isSome);
+    const stakingLedger = stakingLedgerRes.unwrap();
+    assert(stakingLedger.active.eq(boostAmount));
+
+    // Move out of the era we boosted inside of
+    await ExtrinsicHelper.runToBlock(await getNextRewardEraBlock());
+    // Have three eras where we get rewards
+    await ExtrinsicHelper.runToBlock(await getNextRewardEraBlock());
+    await ExtrinsicHelper.runToBlock(await getNextRewardEraBlock());
+    await ExtrinsicHelper.runToBlock(await getNextRewardEraBlock());
+
+    const bobBalance = await getFreeBalance(bob);
+    const aliceBalance = await getFreeBalance(alice);
+
+    const innerClaim = ExtrinsicHelper.api.tx.capacity.claimStakingRewards();
+    const expectedClaimEvent = ExtrinsicHelper.api.events.capacity.ProviderBoostRewardClaimed;
+    await ExtrinsicHelper.proxySignAndSend(innerClaim, alice, bob, expectedClaimEvent);
+
+    // Verify that alice's proxy call cost balance and alice account does not get rewards.
+    const aliceNewBalance = await getFreeBalance(alice);
+    assert(
+      aliceNewBalance < aliceBalance,
+      `expected alice not to get claimed rewards. Alice new - old = ${aliceNewBalance - aliceBalance}`
+    );
+
+    // assert that bob got the claimed rewards.
+    const bobNewBalance = await getFreeBalance(bob);
```

Comment:
the proxy should not get the claimed rewards; bob, the "real" i.e. staker, should.